File: CPlusPlus/10866.cpp
#include <bits/stdc++.h>

#define FASTIO                             \
    std::ios_base::sync_with_stdio(false); \
    std::cin.tie(0);                       \
    std::cout.tie(0);
#define ENDL "\n";

/***************************************************************************************
 *   * 링크 : https://www.acmicpc.net/problem/10866
 ***************************************************************************************/

using namespace std;

int main() {
    FASTIO

    deque<int> dq;

    int testCase, temp;
    cin >> testCase;

    for (int i = 0; i < testCase; i++) {
        string command;
        cin >> command;
        if (command == "push_front") {
            cin >> temp;
            dq.push_front(temp);
        } else if (command == "push_back") {
            cin >> temp;
            dq.push_back(temp);
        } else if (command == "pop_front") {
            if (dq.empty()) {
                cout << "-1" << ENDL;
            } else {
                temp = dq.front();
                cout << temp << ENDL;
                dq.pop_front();
            }
        } else if (command == "pop_back") {
            if (dq.empty()) {
                cout << "-1" << ENDL;
            } else {
                temp = dq.back();
                cout << temp << ENDL;
                dq.pop_back();
            }
        } else if (command == "size") {
            cout << dq.size() << ENDL;
        } else if (command == "empty") {
            if (dq.empty()) {
                cout << "1" << ENDL;
            } else {
                cout << "0" << ENDL;
            }
        } else if (command == "front") {
            if (dq.empty()) {
                cout << "-1" << ENDL;
            } else {
                cout << dq.front() << ENDL;
            }
        } else if (command == "back") {
            if (dq.empty()) {
                cout << "-1" << ENDL;
            } else {
                cout << dq.back() << ENDL;
            }
        }
    }

    return 0;
}

File: BOJ/csharp/10866.cs
using System;
using System.Text;

// 링크 : https://www.acmicpc.net/problem/10866

namespace BOJ
{
    internal class Program
    {
        static void Main()
        {
            StringBuilder sb = new StringBuilder();
            LinkedList<int> list = new LinkedList<int>();

            int cmdCount = int.Parse(Console.ReadLine());

            for (int i = 0; i < cmdCount; i++)
            {
                string[] cmd = Console.ReadLine().Split(' ');

                switch (cmd[0])
                {
                    case "push_front":
                        int num = int.Parse(cmd[1]);
                        list.AddFirst(num);
                        break;

                    case "push_back":
                        num = int.Parse(cmd[1]);
                        list.AddLast(num);
                        break;

                    case "pop_front":
                        if (list.Count != 0)
                        {
                            num = list.First();
                            sb.AppendLine(num.ToString());
                            list.RemoveFirst();
                        }
                        else
                        {
                            sb.AppendLine("-1");
                        }
                        break;

                    case "pop_back":
                        if (list.Count != 0)
                        {
                            num = list.Last();
                            sb.AppendLine(num.ToString());
                            list.RemoveLast();
                        }
                        else
                        {
                            sb.AppendLine("-1");
                        }
                        break;

                    case "size":
                        sb.AppendLine(list.Count.ToString());
                        break;

                    case "empty":
                        if (list.Count == 0)
                        {
                            sb.AppendLine("1");
                        }
                        else
                        {
                            sb.AppendLine("0");
                        }
                        break;

                    case "front":
                        if (list.Count != 0)
                        {
                            num = list.First();
                            sb.AppendLine(num.ToString());
                        }
                        else
                        {
                            sb.AppendLine("-1");
                        }
                        break;

                    case "back":
                        if (list.Count != 0)
                        {
                            num = list.Last();
                            sb.AppendLine(num.ToString());
                        }
                        else
                        {
                            sb.AppendLine("-1");
                        }
                        break;
                }
            }

            Console.Write(sb);
        }
    }
}